### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by gregvanbrug**
Fix some of the eslint errors, added testing framework, set up a few tests as examples.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
     "no-console": 0,
     "vue/component-tags-order": [
       "warning",
       {
-        "order": [
-          "template",
-          "script",
-          "style"
-        ]
+        "order": ["template", "script", "style"]
```
Comment (abea) [inline .eslintrc:23]: Why these changes? It makes it harder to diff.

File: package.json [hunk 2]
```diff
@@ -2,14 +2,36 @@
...
+    "vuedraggable": "^2.23.2"
+  },
+  "devDependencies": {
     "@babel/core": "^7.8.7",
     "@storybook/addon-a11y": "^5.3.17",
     "@storybook/addon-contexts": "^5.3.17",
     "@storybook/addon-knobs": "^5.3.17",
     "@storybook/addon-storysource": "^5.3.17",
+    "@storybook/storybook-deployer": "^2.8.6",
     "@storybook/vue": "^5.3.17",
```
Comment (abea) [inline package.json:29]: @gregvanbrug Are you sure this isn't going to break the deployment since these are now dev dependencies?

File: src/components/areaMenu/AposAreaMenuItem.vue [hunk 3]
```diff
@@ -45,14 +47,14 @@ export default {
 
   methods: {
     click() {
-      this.$emit('click');
+      this.$emit("click");
```
Comment (abea) [inline src/components/areaMenu/AposAreaMenuItem.vue:50]: Let's keep these single quotes.

File: src/components/areaMenu/AposAreaMenuItem.vue [hunk 4]
```diff
@@ -45,14 +47,14 @@ export default {
...
     }
   }
-}
+};
 </script>
 
 <style lang="scss" scoped>
-@import '../../scss/_mixins';
+@import "../../scss/_mixins";
```
Comment (abea) [inline src/components/areaMenu/AposAreaMenuItem.vue:57]: Let's keep these single quotes.

Comment (abea) [reply src/components/areaMenu/AposAreaMenuItem.vue:57]: I'm seeing a ton of these. Is the linter complaining about the single quotes?

Comment (gregvanbrug) [reply src/components/areaMenu/AposAreaMenuItem.vue:57]: Yes. Those were all picked up by eslint.

File: src/components/contextMenu/AposContextMenuItem.vue [hunk 5]
```diff
@@ -12,45 +12,48 @@
...
 <style lang="scss">
-  .apos-context-menu__button {
-    display: inline-block;
-    width: 100%;
-    padding: 10px 20px;
-    border: none;
-    text-align: left;
-    font-size: map-get($font-sizes, default);
-    letter-spacing: 0.5px;
+.apos-context-menu__button {
```
Comment (abea) [inline src/components/contextMenu/AposContextMenuItem.vue:37]: These seem substantial. Was this broken or is there a ticket?

Comment (gregvanbrug) [reply src/components/contextMenu/AposContextMenuItem.vue:37]: I'm working on https://apostrophecms.atlassian.net/jira/software/projects/A30U/boards/4/backlog?selectedIssue=A30U-184. 

With the goal of making a package that can be imported into Apostrophe.

When started storybook, all these linter errors happened. As not all of us have really worked on these components, we should have some tests for the more complex pieces so that when we integrate we're sure the components behave as we expect and saves us headaches if some of these need to be refactored in any way.

Comment (abea) [reply src/components/contextMenu/AposContextMenuItem.vue:37]: Okay. This seems to actually change styles, though. Unclear why that changed.

File: src/components/areaMenu/AposAreaMenuItem.vue (hunk 4 above)
Comment (abea) [reply src/components/areaMenu/AposAreaMenuItem.vue:57]: From running `npm test` here? I'm unable to get the single quotes to throw an error. This is generally not our style.

File: package.json (hunk 2 above)
Comment (gregvanbrug) [reply package.json:29]: We're running deploys locally so there shouldn't be an issue. All the @storybook modules are moved to dev dependencies so they don't install when this module is installed elsewhere as we'll only need the components not storybook itself.

File: src/components/contextMenu/AposContextMenuItem.vue (hunk 5 above)
Comment (gregvanbrug) [reply src/components/contextMenu/AposContextMenuItem.vue:37]: It appears my editor got a little aggressive trying to "fix" things. Not sure why it's not following our rules in this repository... Would be nice though if we could use prettier or run eslint fix on a precommit hook or something so we don't need to think about formatting issues like this.

Comment (abea) [reply src/components/contextMenu/AposContextMenuItem.vue:37]: Since this project was not linted at all before, I disabled auto fix in VS Code for the project to make sure I know what I was changing. Running eslint fix on precommit seems alright. No requirement to have prettier installed locally. It seems a little weird to not see what you're  committing before you do it, but should be fine.

Comment (gregvanbrug) [reply src/components/contextMenu/AposContextMenuItem.vue:37]: Ok. So do you want me to roll back all the formatting stuff and just commit the testing framework and make another PR for linting? It probably should play by the rules, right?

Comment (abea) [reply src/components/contextMenu/AposContextMenuItem.vue:37]: That might be easier. Keeping the PR focused.
